test/acceptor_SUITE.erl

```diff
@@ -532,6 +533,35 @@ ssl_echo(_) ->
 	{'EXIT', _} = begin catch ranch:get_port(Name) end,
 	ok.
 
+ssl_handshake(_) ->
+	doc("Ensure that multi-step handshake works with SSL transport."),
+	Name = name(),
+	{CaCert1, Cert1, Key1} = ct_helper:make_certs(),
+	{CaCert2, Cert2, Key2} = ct_helper:make_certs(),
+	Opts1 = [{cert, Cert1}, {key, Key1}, {cacerts, [CaCert1]}, {verify, verify_peer}],
+	Opts2 = [{cert, Cert2}, {key, Key2}, {cacerts, [CaCert2]}, {verify, verify_peer}],
+	DefaultOpts = ct_helper:get_certs_from_ets(),
+	{ok, _} = ranch:start_listener(Name,
+		ranch_ssl, [{handshake, hello}|DefaultOpts],
```

Comment:
OK, it would be good if we didn't have to configure this in the transport options, but rather that the protocol module itself could configure it. Can it be passed to `ssl:handshake` directly? If not, then we need to think about another way to signal that we want hello that doesn't involve the user of a protocol module setting it explicitly.

Reply:
> Can it be passed to ssl:handshake directly?

According to the `ssl` documentation, no, only `server_options()` can be given here, and the `handshake` option is not among those. I tried, and it turns out that the documentation is indeed correct.

Reply:
I'll have to think about if we could, in theory, emulate such a behavior, but it's not possible with what `ssl` provides out of the box.

Reply:
I don't think we can emulate, and all the solutions I think of are unsatisfactory. Probably fine to leave as-is for now. Then protocol modules can support both with and without the option I guess.